pkg/sharing/secret_reconciler.go

```diff
@@ -0,0 +1,178 @@
+package sharing
```

Comment:
in general,  ask your doc if godocs are right for each function or exported type below :)